src/daft-io/src/azure_blob.rs

```diff
@@ -64,6 +65,52 @@ enum Error {
     NotAFile { path: String },
 }
 
+/// Parse an Azure URI into its components.
+/// Returns (protocol, container if exists, key).
+fn parse_azure_uri(uri: &str) -> super::Result<(String, Option<String>, String)> {
```

Comment:
you shouldn't need to allocate new strings here but can rely on returning `&'a str` for each of these.